tools/src/main/java/org/apache/kafka/tools/DelegationTokenCommand.java

```diff
@@ -0,0 +1,306 @@
+        List<KafkaPrincipal> ownerPrincipals = getPrincipals(opts, opts.ownerPrincipalsOpt);
+
+        if (ownerPrincipals.isEmpty()) {
+            System.out.println("Calling describe token operation for current user.");
+        } else {
+            System.out.println("Calling describe token operation for owners :" + ownerPrincipals);
+        }
+
+        DescribeDelegationTokenResult describeResult = adminClient.describeDelegationToken(new DescribeDelegationTokenOptions().owners(ownerPrincipals));
+        List<DelegationToken> tokens = describeResult.delegationTokens().get();
+        System.out.printf("Total number of tokens : %d", tokens.size());
+        printToken(tokens);
+        return tokens;
+    }
+
+    private static Admin createAdminClient(DelegationTokenCommandOptions opts) throws IOException {
+        Properties props = Utils.loadProps(opts.options.valueOf(opts.commandConfigOpt));
+        props.put("bootstrap.servers", opts.options.valueOf(opts.bootstrapServerOpt));
+        return Admin.create(props);
+    }
+
+    static class DelegationTokenCommandOptions extends CommandDefaultOptions {
+        public final ArgumentAcceptingOptionSpec<String> bootstrapServerOpt;
+        public final ArgumentAcceptingOptionSpec<String> commandConfigOpt;
+        public final AbstractOptionSpec<Void> createOpt;
+        public final AbstractOptionSpec<Void> renewOpt;
+        public final AbstractOptionSpec<Void> expiryOpt;
+        public final AbstractOptionSpec<Void> describeOpt;
+        public final ArgumentAcceptingOptionSpec<String> ownerPrincipalsOpt;
+        public final ArgumentAcceptingOptionSpec<String> renewPrincipalsOpt;
+        public final ArgumentAcceptingOptionSpec<Long> maxLifeTimeOpt;
+        public final ArgumentAcceptingOptionSpec<Long> renewTimePeriodOpt;
+        public final ArgumentAcceptingOptionSpec<Long> expiryTimePeriodOpt;
+        public final ArgumentAcceptingOptionSpec<String> hmacOpt;
+
+        public DelegationTokenCommandOptions(String[] args) {
+            super(args);
+
+            String bootstrapServerDoc = "REQUIRED: server(s) to use for bootstrapping.";
+            String commandConfigDoc = "REQUIRED: A property file containing configs to be passed to Admin Client. Token management" +
+                    " operations are allowed in secure mode only. This config file is used to pass security related configs.";
+
+            this.bootstrapServerOpt = parser.accepts("bootstrap-server", bootstrapServerDoc)
+                    .withRequiredArg()
+                    .ofType(String.class);
+
+            this.commandConfigOpt = parser.accepts("command-config", commandConfigDoc)
+                    .withRequiredArg()
+                    .ofType(String.class);
+
+            this.createOpt = parser.accepts("create", "Create a new delegation token. Use --renewer-principal option to pass renewers principals.");
```

Comment:
suggestion
            this.createOpt = parser.accepts("create", "Create a new delegation token. Use --renewer-principal option to pass renewer principals.");
```
```